src/pages/home/report/withReportAndReportActionOrNotFound.js

```diff
@@ -0,0 +1,153 @@
+import NotFoundPage from '../../ErrorPage/NotFoundPage';
+import ONYXKEYS from '../../../ONYXKEYS';
+import reportPropTypes from '../../reportPropTypes';
+import reportActionPropTypes from './reportActionPropTypes';
+import FullscreenLoadingIndicator from '../../../components/FullscreenLoadingIndicator';
+import * as ReportUtils from '../../../libs/ReportUtils';
+import * as ReportActionsUtils from '../../../libs/ReportActionsUtils';
+import * as Report from '../../../libs/actions/Report';
+import compose from '../../../libs/compose';
+import withWindowDimensions from '../../../components/withWindowDimensions';
+
+export default function (WrappedComponent) {
+    const propTypes = {
+        /** The HOC takes an optional ref as a prop and passes it as a ref to the wrapped component.
+         * That way, if a ref is passed to a component wrapped in the HOC, the ref is a reference to the wrapped component, not the HOC. */
+        forwardedRef: PropTypes.func,
+
+        /** The report currently being looked at */
+        report: reportPropTypes,
+
+        /** Array of report actions for this report */
+        reportActions: PropTypes.shape(reportActionPropTypes),
+
+        /** The policies which the user has access to */
+        policies: PropTypes.objectOf(
+            PropTypes.shape({
+                /** The policy name */
+                name: PropTypes.string,
+
+                /** The type of the policy */
+                type: PropTypes.string,
+            }),
+        ),
+
+        /** Route params */
+        route: PropTypes.shape({
+            params: PropTypes.shape({
+                /** Report ID passed via route */
+                reportID: PropTypes.string,
+
+                /** ReportActionID passed via route */
+                reportActionID: PropTypes.string,
+            }),
+        }).isRequired,
+
+        /** Beta features list */
+        betas: PropTypes.arrayOf(PropTypes.string),
+
+        /** Indicated whether the report data is loading */
+        isLoadingReportData: PropTypes.bool,
+
+        /** Is the window width narrow, like on a mobile device? */
+        isSmallScreenWidth: PropTypes.bool.isRequired,
+    };
+
+    const defaultProps = {
+        forwardedRef: () => {},
+        reportActions: {},
+        report: {},
+        policies: {},
+        betas: [],
+        isLoadingReportData: true,
+    };
+
+    // eslint-disable-next-line rulesdir/no-negated-variables
+    function WithReportAndReportActionOrNotFound(props) {
+        // For small screen, we don't call openReport API when we go to a sub report page by deeplink
+        // So we need to call openReport here for small screen
+        useEffect(() => {
+            if (!props.isSmallScreenWidth || !_.isEmpty(props.report)) {
+                return;
+            }
+            Report.openReport(props.route.params.reportID);
+            // eslint-disable-next-line react-hooks/exhaustive-deps
+        }, []);
```

Comment:
Why would we not want to depend on the `props.isSmallScreenWdith`  and `props.report` ?